addon/actions.js

```diff
@@ -0,0 +1,55 @@
+import _ from 'lodash'
+import {validateValue} from './validator/index'
+import {aggregateResults} from './validator/utils'
+import {getPath} from './dereference'
+
+export const CHANGE_VALUE = 'CHANGE_VALUE'
+export const VALIDATION_RESOLVED = 'VALIDATION_RESOLVED'
+
+export function changeValue (bunsenId, value) {
+  return {
+    type: CHANGE_VALUE,
+    bunsenId,
+    value
+  }
+}
+
+export function updateValidationResults (validationResult) {
+  // Group errors by input
+  let errors = _.groupBy(validationResult.errors, 'path')
+
+  // Filter errors to just be messages not complex objects
+  errors = _.mapValues(errors, (fieldErrors, bunsenId) => _.pluck(fieldErrors, 'message'))
```

Comment:
it's okay to create new `const` objects as you go in order for this to be more self-explanatory, too, for example:
```
const errorsByInput = _.groupBy(errorsList, inputSelector)
const errorsFilteredToMessagesOnly = _.mapValues(errorsByInput, ...)
const errorsMappedToDotNotation = _.mapKeys(errorsFilteredToMessagesOnly, ...)

return {
  errors: errorsMappedToDotNotation,
  ...
}
```